### PR TITLE
feat: 🍰 Add The Languages French And Dutch

### LINKED ISSUE #2135: 🚀 [Feature] add the languages french and dutch
## 🚀 Feature
add the languages french and dutch

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
add locales french and dutch

### Issues
- fixes #2135 
 

image

### CODE REVIEW

**Comment by BerndHueckstaedt** [comment]
I had understood Christopher's design so that the current language is on the left and the other languages can be selected in a drop-down menu. This is what the triangle was meant for.

ich hatte den Entwurf von Christopher so verstanden, dass die aktuelle Sprache links steht und die jeweils anderen Sprachen in einem Aufklapp-Menü ausgewählt werden können. Dafür war das Dreieck gedacht.

**Review by Mogge — CHANGES_REQUESTED** [review]
Please clean up. Remove everything that is not related lo locales.

**Comment by ogerly** [comment]
> Please clean up. Remove everything that is not related lo locales.

oh ... somehow the statistics got in there

**Comment by ogerly** [comment]
> I had understood Christopher's design so that the current language is on the left and the other languages can be selected in a drop-down menu. This is what the triangle was meant for.
> 
> ich hatte den Entwurf von Christopher so verstanden, dass die aktuelle Sprache links steht und die jeweils anderen Sprachen in einem Aufklapp-Menü ausgewählt werden können. Dafür war das Dreieck gedacht.

is changed

File: frontend/src/locales/fr.json [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      "gddSendAmount": "L´espace {_field_} doit comprendre un nombre entre {min} et {max} avec un maximum de deux chiffres après la virgule",
+      "is-not": "Vous ne pouvez pas vous envoyer de Gradido à vous-même",
+      "usernmae-regex": "Le nom d´utilisateur doit commencer par une lettre, suivi d´au moins deux caractères alphanumériques.",
+      "usernmae-unique": "Ce nom d´utilisateur est déjà pris."
+    },
+    "your_amount": "Votre montant"
+  },
+  "GDD": "GDD",
+  "gdd_per_link": {
+    "choose-amount": "Sélectionnez le montant que vous souhaitez envoyer via le lien. Vous pouvez également joindre un message. Cliquez sur ‘créer maintenant’ pour établir un lien que vous pourrez partager.",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:155]:
```suggestion
    "choose-amount": "Sélectionnez le montant que vous souhaitez envoyer via lien. Vous pouvez également joindre un message. Cliquez sur ‘créer maintenant’ pour établir un lien que vous pourrez partager.",
```

File: frontend/src/locales/fr.json [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    "copy-link": "Copier le lien",
+    "copy-link-with-text": "Copier le lien et le texte",
+    "created": "Le lien a été créé!",
+    "credit-your-gradido": "Pour l´accréditation du Gradido, cliquer sur le lien!",
+    "decay-14-day": "Perte sur 14 jours",
+    "delete-the-link": " Supprimer le lien?",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:161]:
```suggestion
    "delete-the-link": "Supprimer le lien?",
```

File: frontend/src/locales/fr.json [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    "has-account": "Vous avez déjà un compte Gradido?",
+    "header": "Envoyer des Gradidos via lien",
+    "isFree": "Gradido est gratuit mondialement.",
+    "link-and-text-copied": "Le lien et votre message ont été copiés dans le presse-papier. Vous pouvez maintenant le joindre à un email ou à un message..",
+    "link-copied": "Le lien a été copié dans le presse-papier. Vous pouvez désormais le coller dans votre email ou votre message.",
+    "link-deleted": "Le lien a été supprimé le on {date}.",
+    "link-expired": "Le lien n´est plus valide. Sa validité a expiré le {date}.",
+    "link-overview": "Aperçu du lien",
+    "links_count": "Liens actifs",
+    "links_sum": "Ouvrir les liens et les codes QR",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:173]:
```suggestion
    "links_sum": "Ouvrir les liens et les QR codes",
```

File: frontend/src/locales/fr.json [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+    "no-account": "Vous n´avez pas encore de compte Gradido?",
+    "no-redeem": "Vous n´êtes pas autorisé à percevoir votre propre lien!",
+    "not-copied": "Malheureusement votre appareil ne permet pas de copier ! Veuillez copier le lien  manuellement svp!",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:176]:
```suggestion
    "not-copied": "Malheureusement votre appareil ne permet pas de copier! Veuillez copier le lien manuellement svp!",
```

File: frontend/src/locales/fr.json [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+    "redeem": " Encaisser",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:177]:
```suggestion
    "redeem": "Encaisser",
```

File: frontend/src/locales/fr.json [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+    "redeem-text": "Voulez-vous percevoir le montant maintenant?",
+    "redeemed": "Encaissé avec succès ! Votre compte est crédité de {n} GDD.",
+    "redeemed-at": " Le lien a déjà été perçu le {date}.",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:180]:
```suggestion
    "redeemed-at": "Le lien a déjà été perçu le {date}.",
```

File: frontend/src/locales/fr.json [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+    "redeemed-title": "encaisser ",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:181]:
```suggestion
    "redeemed-title": "encaisser",
```

File: frontend/src/locales/fr.json [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+    "link-expired": "Le lien n´est plus valide. Sa validité a expiré le {date}.",
+    "link-overview": "Aperçu du lien",
+    "links_count": "Liens actifs",
+    "links_sum": "Ouvrir les liens et les codes QR",
+    "no-account": "Vous n´avez pas encore de compte Gradido?",
+    "no-redeem": "Vous n´êtes pas autorisé à percevoir votre propre lien!",
+    "not-copied": "Malheureusement votre appareil ne permet pas de copier ! Veuillez copier le lien  manuellement svp!",
+    "redeem": " Encaisser",
+    "redeem-text": "Voulez-vous percevoir le montant maintenant?",
+    "redeemed": "Encaissé avec succès ! Votre compte est crédité de {n} GDD.",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:179]:
```suggestion
    "redeemed": "Encaissé avec succès! Votre compte est crédité de {n} GDD.",
```

File: frontend/src/locales/fr.json [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+    "community": "Communauté",
+    "logout": "Déconnexion",
+    "members_area": "Partie réservée aux membres",
+    "overview": "Aperçu",
+    "profile": "Mon profile",
+    "send": "Envoyer",
+    "support": "Aide",
+    "transactions": "Transactions"
+  },
+  "qrCode": "Code QR",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:235]:
```suggestion
  "qrCode": "QR Code",
```

File: frontend/src/locales/fr.json [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+  "send_gdd": "Envoyer GDD",
+  "send_per_link": "Envoyer GDD via lien",
+  "session": {
+    "extend": "Rester connecter",
+    "lightText": "S´il n´apparaît aucune activité   pendant plus de 10 minutes, la session expirera pour des raisons de sécurité.",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:240]:
```suggestion
    "lightText": "S´il n´apparaît aucune activité pendant plus de 10 minutes, la session expirera pour des raisons de sécurité.",
```

File: frontend/src/locales/fr.json [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+    "logoutIn": "Se déconnecter ",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:241]:
```suggestion
    "logoutIn": "Se déconnecter",
```

**Review by Elweyn — CHANGES_REQUESTED** [review]

File: frontend/src/locales/de.json [hunk 12]
```diff
@@ -247,6 +247,8 @@
       "de": "Deutsch",
       "en": "English",
       "es": "Español",
+      "fr": "French",
```
Comment (Mogge) [inline frontend/src/locales/de.json:250]:
```suggestion
      "fr": "Français",
```

File: frontend/src/locales/en.json [hunk 13]
```diff
@@ -246,7 +246,9 @@
       "changeLanguage": "Change language",
       "de": "Deutsch",
       "en": "English",
-      "es": "Español",
+      "es": "Spanish",
+      "fr": "French",
```
Comment (Mogge) [inline frontend/src/locales/en.json:250]:
```suggestion
      "fr": "Français",
```

File: frontend/src/locales/en.json [hunk 14]
```diff
@@ -246,7 +246,9 @@
       "changeLanguage": "Change language",
       "de": "Deutsch",
       "en": "English",
-      "es": "Español",
+      "es": "Spanish",
```
Comment (Mogge) [inline frontend/src/locales/en.json:249]:
```suggestion
      "es": "Español",
```

File: frontend/src/locales/en.json [hunk 15]
```diff
@@ -246,7 +246,9 @@
...
+      "fr": "French",
+      "nl": "Dutch",
```
Comment (Mogge) [inline frontend/src/locales/en.json:251]:
```suggestion
      "nl": "Holandés",
```

File: frontend/src/locales/index.js [hunk 16]
```diff
@@ -17,6 +17,18 @@ const locales = [
     iso: 'es-ES',
     enabled: true,
   },
+  {
+    name: 'French',
```
Comment (Mogge) [inline frontend/src/locales/index.js:21]:
```suggestion
    name: 'Français',
```

File: frontend/src/locales/index.js [hunk 17]
```diff
@@ -17,6 +17,18 @@ const locales = [
...
+    code: 'fr',
+    iso: 'fr-FR',
+    enabled: true,
+  },
+  {
+    name: 'Dutch',
```
Comment (Mogge) [inline frontend/src/locales/index.js:27]:
```suggestion
    name: 'Holandés',
```

**Review by Mogge — CHANGES_REQUESTED** [review]

**Review by Tirokk — CHANGES_REQUESTED** [review]
Changes in NL …

**Review by Tirokk — CHANGES_REQUESTED** [review]
I would refine the design a bit, see at the end.

And there are kind of design errors as well.
In `Holandés` the pipe between the choosen language and the arrow is missing. And there is a pipe to much after the last language if you open the menu:

Bildschirmfoto 2022-08-17 um 10 45 25

I would consider:

- to completely remove the pipe between the choosen language and the arrow and put the arrow directly aside the lange and have both clickable to open the menu.
- or to have english always additional on the right side of the pipe. Then have a second pipe with `Others` and the arrow aside it and have both clickable to oppen the menu.

Bildschirmfoto 2022-08-17 um 10 44 16

**Comment by ogerly** [comment]
@Tirokk i have adjusted this. however, when i select dutch hiter french is a pipe ... this is a bit more complicated than i thought. i need to look at this with more musse. otherwise it fits quite well now ... 
https://user-images.githubusercontent.com/1324583/185113267-f1e93f1a-a772-44e2-a098-6b7dd2c1041d.mp4

**Review by Tirokk — DISMISSED** [review]
Yes !!!

I approve 🚀🚀💫💫

**Comment by ogerly** [comment]
close menu when selected another language

https://user-images.githubusercontent.com/1324583/185453161-587fd123-93fc-4bd8-9d53-87281a8516d9.mp4

File: frontend/src/locales/fr.json [hunk 18]
```diff
@@ -300,6 +307,14 @@
       "uppercase": "Une lettre majuscule requise."
     }
   },
+  "statistic": {
+    "activeUsers": "Membres actifs",
+    "deletedUsers": "Membres supprimés",
+    "totalGradidoAvailable": "GDD total en circulation",
+    "totalGradidoCreated": "GDD total épuisé",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:314]: Épuisé bedeutet erschöpft! Soll aber wohl geschöpft heißen oder? Wenn ja dann:
```suggestion
    "totalGradidoCreated": "GDD total puisé",
```

File: frontend/src/locales/fr.json [hunk 19]
```diff
@@ -300,6 +307,14 @@
...
+    "totalGradidoDecayed": "GDD total annulé",
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:315]:
```suggestion
    "totalGradidoDecayed": "Total de GDD écoulé",
```

File: frontend/src/locales/fr.json [hunk 20]
```diff
@@ -300,6 +307,14 @@
...
+    "totalGradidoUnbookedDecayed": "Total GDD non comptabilisé Expiration"
```
Comment (Elweyn) [inline frontend/src/locales/fr.json:316]:
```suggestion
    "totalGradidoUnbookedDecayed": "Total GDD non comptabilisé écoulé"
```

**Review by Elweyn — CHANGES_REQUESTED** [review]

**Review by Elweyn — APPROVED** [review]

**Review by Mogge — APPROVED** [review]

**Pull request merged**